src/property.ts

```diff
@@ -927,7 +927,7 @@ function Property$ensureInited(property: Property, obj: Entity): void {
 
 			const underlyingValue = obj.__fields__[property.name];
 			// Mark the property as pending initialization if it still has no underlying value, or is the property type's default, to allow default calculation rules to run for it
-			if (underlyingValue === property.defaultValue || (property.isList && Array.isArray(underlyingValue) && underlyingValue.length === 0))
+			if (underlyingValue === property.defaultValue)
```

Comment:
Discussed this with Bryan, and determined this logic is a bit confusing, since it is avoiding defaulting lists in a very subtle way. Suggestion is to add an explicit check for array, and comment:

```ts
// List properties are defaulted onInitNew instead of on access
if (underlyingValue === property.defaultValue && !Array.isArray(underlyingValue))
```

@mattheyan Feel free to comment with any additional feedback.

Reply:
Updated